pkg/apiserver/apiserver.go

```diff
@@ -41,7 +41,7 @@ func init() {
 	eminstall.Install(Scheme)
 
 	// we need custom conversion functions to list resources with options
-	installer.RegisterConversions(Scheme)
+	installer.RegisterConversions(Scheme) //nolint: errcheck
```

Comment:
Let's add a todo here to fix this errcheck, not checking errors is pretty bad.

More generally, this piece of code shouldn't be in a `init()` function, also it installs both custom-metrics and external-metrics every time even though some projects might only want to import one API.

Reply:
Ok. I added a nolint exception because I thought it was the same thing that was all over Kubernetes with these  `zz_generated.conversion.go` files (e.g. [here](https://github.com/kubernetes/kubernetes/blob/c1e69551be1a72f0f8db6778f20658199d3a686d/plugin/pkg/admission/podtolerationrestriction/apis/podtolerationrestriction/v1alpha1/zz_generated.conversion.go#L35)), but with a closer look, it's not.

I agree that this code needs to be reworked because custom-metrics and external-metrics should not be both statically installed unconditionally.

Reply:
Oh correct, I didn't check what `RegisterConversions` was doing when commenting but now that you've explained that is written the same way as `zz_generated.conversion.go` generated by the code-generator from k8s, I am fine with keeping the exception.

Reply:
I switched to `k8s.io/apimachinery/pkg/util/runtime.Must()`, which is used in this kind of cases in Kubernetes. It would panic the same way in case of error, but at least it is not linter-specific.